MinnPost/object-sync-for-salesforce
# Remove the requirement to set the Salesforce REST API version from the interface in favor of a plugin-wide setting.

## What does this Pull Request do?

This fixes #458 by removing the requirement to set the Salesforce REST API version from the interface in favor of a plugin-wide setting. This setting allows us to track Salesforce API releases within the plugin itself rather than putting that responsibility onto users.

For backward-compatibility, for the time being the plugin will use previously-provided settings (both as config values and as `wp_options` values) and show a notice to users if they are using one of those settings.

For future developer flexibility, this also adds a developer hook in case developers want to set their own version of the API.

## How do I test this Pull Request?

- use the plugin without any previously set version: it should use the value from the object-sync-for-salesforce.php file (currently 55.0).
- use the plugin with a previously set config value that is not 55.0
- use the plugin with a previously set `wp_options` value that is not 55.0
- use the plugin with a developer hook value that is not 55.0.
